shablona/tests/test_shablona.py

```diff
@@ -1,110 +1,35 @@
-from __future__ import absolute_import, division, print_function
-import os.path as op
+import pytest
 import numpy as np
-import pandas as pd
-import numpy.testing as npt
-import shablona as sb
 
-data_path = op.join(sb.__path__[0], 'data')
+import shablona
 
+def test_midi_to_hz_float():
+    expected = 440.0
+    assert shablona.midi_to_hz(69) == expected
 
-def test_transform_data():
-    """
-    Testing the transformation of the data from raw data to functions
-    used for fitting a function.
+def test_midi_to_hz_array():
+    expected = [261.6255653, 329.62755691, 440.0]
+    assert np.allclose(shablona.midi_to_hz([60, 64, 69]),  expected)
 
-    """
-    # We start with actual data. We test here just that reading the data in
-    # different ways ultimately generates the same arrays.
-    from matplotlib import mlab
-    ortho = mlab.csv2rec(op.join(data_path, 'ortho.csv'))
-    x1, y1, n1 = sb.transform_data(ortho)
-    x2, y2, n2 = sb.transform_data(op.join(data_path, 'ortho.csv'))
-    npt.assert_equal(x1, x2)
-    npt.assert_equal(y1, y2)
-    # We can also be a bit more critical, by testing with data that we
-    # generate, and should produce a particular answer:
-    my_data = pd.DataFrame(
-        np.array([[0.1, 2], [0.1, 1], [0.2, 2], [0.2, 2], [0.3, 1],
-                  [0.3, 1]]),
-        columns=['contrast1', 'answer'])
-    my_x, my_y, my_n = sb.transform_data(my_data)
-    npt.assert_equal(my_x, np.array([0.1, 0.2, 0.3]))
-    npt.assert_equal(my_y, np.array([0.5, 0, 1.0]))
-    npt.assert_equal(my_n, np.array([2, 2, 2]))
+def test_hz_to_midi_float():
+    expected = 69
+    assert shablona.hz_to_midi(440.0) == expected
 
+def test_hz_to_midi_array():
+    expected = [57, 69, 81]
+    assert np.allclose(shablona.hz_to_midi([220.0, 440.0, 880.0]), expected)
 
-def test_cum_gauss():
-    sigma = 1
-    mu = 0
-    x = np.linspace(-1, 1, 12)
-    y = sb.cumgauss(x, mu, sigma)
-    # A basic test that the input and output have the same shape:
-    npt.assert_equal(y.shape, x.shape)
-    # The function evaluated over items symmetrical about mu should be
-    # symmetrical relative to 0 and 1:
-    npt.assert_equal(y[0], 1 - y[-1])
-    # Approximately 68% of the Gaussian distribution is in mu +/- sigma, so
-    # the value of the cumulative Gaussian at mu - sigma should be
-    # approximately equal to (1 - 0.68/2). Note the low precision!
-    npt.assert_almost_equal(y[0], (1 - 0.68) / 2, decimal=2)
+# Hello!  You could add the missing test here, in Part 5!
```

Comment:
it'd be good to give the name of the test function here, eg "You could add test_midi_to_hz here" or whatever it is.  It would help cut down the things people have to worry about.

Reply:
🚀